### DISCUSSION TITLE
Format of image in ring/+/camera/+/snapshot/image

### DISCUSSION THREAD

**Discussion by irvintim** [tsightler/ring-mqtt#419] [Support Q&A] [answered]
I can't figure out what the format is for the image that is sent in the MQTT topic "ring/<id>/camera/<dev>/snapshot/image".  It seems to be a unicoded binary file, so I have decoded it back to binary but it's not readable as bmp, jpg, png, gif, ...  I tried to use ImageMagick to identify it but that gave me "identify: no decode delegate for this image format". 

Can someone tell me how I can decode this unicode blob back to an image file?

Thanks!

Tim

**Comment by tsightler** [comment, marked as answer]
It's JPEG for sure.  I wouldn't expect it to be encoded at all as I don't do any manipulation of the binary data before sending it, literally the exact binary image data received from the Ring API is what is sent directly via MQTT.

**Reply by irvintim** [reply]
Confirmed, it is JPEG.  Apparently the AWS MQTT Test Client was doing the extra encoding and breaking the image, as soon as I started to save the data directly from the MQTT message instead of grabbing it from the Test Client it works as a JPEG image.  Sorry about the stupid question, thanks for getting back to me.
